### PR TITLE
misc: do not publish lighthouse-cli/test except smokehouse

### PR DESCRIPTION
**Pull request by connorjclark**
ref #8084

image

afterthought ... would this break something in publisher-ads? maybe we should keep the smokerunner files but get rid of the fixtures n stuff?

### CODE REVIEW

File: lighthouse-core/scripts/release/test.sh [hunk 1]
```diff
@@ -35,8 +35,9 @@ mkdir -p /tmp/lighthouse-local-test
 cd /tmp/lighthouse-local-test
 
 npm init -y
-npm install "$LH_PRISTINE_ROOT/lighthouse-$VERSION.tgz"
-cd node_modules/lighthouse/lighthouse-cli/test/ && npm install lodash.clonedeep && cd ../../../../
```
Comment (connorjclark) [inline lighthouse-core/scripts/release/test.sh:39]: this command was accidentally installing all our devDeps

idk why I had it cd to that particular folder 🤣

**Comment by brendankenny** [comment]
> afterthought ... would this break something in publisher-ads?

yes: https://github.com/googleads/publisher-ads-lighthouse-plugin/blob/master/lighthouse-plugin-publisher-ads/test/smoke/run-smoke.js#L50

**Comment by brendankenny** [comment]
(they use all their own tests, though)

**Comment by connorjclark** [comment]
keeping just the code is an extra ~50KB ✅ 
image

File: .npmignore [hunk 2]
```diff
@@ -22,8 +22,9 @@ lighthouse-logger/
 !lighthouse-core/scripts/manual-chrome-launcher.js
 !lighthouse-core/scripts/download-chrome.sh
 
-# keep smokehouse tests, etc
-!lighthouse-cli/test
```
Comment (patrickhulce) [inline .npmignore:26]: was this a noop previously? it seems like we never actually said to ignore this folder, so why did we negate it... 🤔

File: lighthouse-cli/test/smokehouse/frontends/smokehouse-bin.js [hunk 3]
```diff
@@ -19,7 +19,12 @@ const log = require('lighthouse-logger');
 const {runSmokehouse} = require('../smokehouse.js');
 const {server, serverForOffline} = require('../../fixtures/static-server.js');
 
-const coreTestDefnsPath = require.resolve('../test-definitions/core-tests.js');
+let coreTestDefnsPath = '';
```
Comment (patrickhulce) [inline lighthouse-cli/test/smokehouse/frontends/smokehouse-bin.js:22]: does `path.join(__dirname, '../test-definitions/core-tests.js')` work instead? this try/catch feels a bit awkward

File: .npmignore [hunk 4]
```diff
@@ -22,8 +22,9 @@ lighthouse-logger/
...
+lighthouse-cli/test/cli
```
Comment (patrickhulce) [inline .npmignore:25]:
```suggestion
# Exclude the CLI smoketests but keep the smoketest runner that is used by downstream projects (e.g. publisher ads).
```

maybe?

**Review by patrickhulce — APPROVED** [review]

File: lighthouse-cli/test/smokehouse/frontends/smokehouse-bin.js (hunk 3 above)
Comment (connorjclark) [reply lighthouse-cli/test/smokehouse/frontends/smokehouse-bin.js:22]: yes but i was opting to not change the string value.

File: .npmignore (hunk 2 above)
Comment (brendankenny) [reply .npmignore:26]: I think it was just to make it explicit, more important when release was just a checklist, not a script, so missing tests might not be noticed.

**Comment by brendankenny** [comment]
@connorjclark any chance you could land this?

Looking at the publisher-ads use of smokehouse in https://github.com/GoogleChrome/lighthouse/pull/12415#issuecomment-828016795 I had an idea to make it less ugly, but it would bump into this PR. Plus it's always a good time to trim our install size :)

**Pull request merged**